js/components/rubric-summary.js

```diff
@@ -0,0 +1,86 @@
+import React, { PureComponent } from 'react'
+import colorScale from '../util/colors'
+import '../../css/rubric-summary.less'
+import RubricBox from './rubric-box'
+export default class RubricSummary extends PureComponent {
+  constructor (props) {
+    super(props)
+    this.state = {
+      showLightbox: false
+    }
+  }
+
+  openLightBox (event) {
+    console.log('open')
+    this.setState({showLightbox: true})
+  }
+
+  closeLightBox (event) {
+    console.log('close')
+    event.stopPropagation()
+    this.setState({showLightbox: false})
+  }
+
+  renderLightBox () {
+    const {rubric} = this.props
+    const rowMaps = this.getRowMaps()
+    return (
+      <div className='rubric-lightbox-wrapper'>
+        <div className='rubric-lightbox'>
+          <div className='closebox' onClick={this.closeLightBox.bind(this)}>✖</div>
```

Comment:
`bind` will return a new function each time you call it. It means that every time you render parent component, this child component will have some property changed and it will propagate react invalidation / re-rendering machinery further down, even if it's not necessary. In most cases it's not very imporant, but if the given child component is expensive to render, it might make some difference (here it's just a DIV, so whatever). Or if we have multiple / hundrets of cases like this (when you render some list). Same with dynamically defined handlers like `onClick={ (e) => doSomething(e) }`.